src/core/public/chrome/nav_links/nav_links_service.ts

```diff
@@ -163,3 +169,19 @@ function sortNavLinks(navLinks: ReadonlyMap<string, NavLinkWrapper>) {
     'order'
   );
 }
+
+function toNavDeepLinks(
+  app: PublicAppInfo,
+  deepLinks: PublicAppDeepLinkInfo[],
+  http: HttpStart
+): Array<[string, NavLinkWrapper]> {
+  if (!deepLinks) return [];
+  return deepLinks.reduce((navDeepLinks: Array<[string, NavLinkWrapper]>, deepLink) => {
+    const id = `${app.id}:${deepLink.id}`;
```

Comment:
This composed id was needed to ensure the navLink id of a deepLink is unique across all applications.

Reply:
No, we are not parsing this id back anywhere, this is just to prevent deep link id collisions between apps.